Import/AdditionalServiceImport.sql

```diff
@@ -0,0 +1,142 @@
+
+IF type_id('[dbo].[AdditionalServiceImportCapability]') IS NULL
+CREATE TYPE [dbo].[AdditionalServiceImportCapability] as TABLE (CapabilityRef varchar(10)) 
+GO
+
+CREATE OR ALTER PROCEDURE AddtionServiceImport
+	@SupplierId varchar(6),
+	@ParentSolutionId varchar(16),
+	@AdditionServiceId varchar(16),
+	@ServiceName varchar(255),
+	@ServiceSummary varchar(300),
+	@ServiceDescription varchar(3000),
+	@Capabilities AdditionalServiceImportCapability READONLY   
+
+AS  
+BEGIN TRAN
+
+BEGIN TRY
+
+	/***
+	Setup
+	***/
+	DECLARE @EmptyGuid UNIQUEIDENTIFIER = 0x0
+	DECLARE @SolutionDetailId UNIQUEIDENTIFIER
+	DECLARE @PublishedStatusId int
+	DECLARE @AuthorityStatusId int
+	DECLARE @SupplierStatusId int
+
+	/***
+	Verify that Parent Solution ID is correct for Additional Service ID
+	***/
+	IF SUBSTRING(@ParentSolutionId, 1, CHARINDEX('-', @ParentSolutionId) - 1) <> @SupplierId
+		THROW 51000, 'Parent Solution ID not correct for Supplier.', 1;  
+
+	/***
+	Verify that Parent Solution ID is correct for Additional Service ID
+	***/
+	IF SUBSTRING(@AdditionServiceId, 1, CHARINDEX('A', @AdditionServiceId) - 1) <> @ParentSolutionId
+		THROW 51000, 'Additional Service ID not correct for parent Solution ID.', 1;  
+
+	/***
+	Verify that parent Solution Exists
+	***/
+	IF NOT EXISTS (SELECT (1) FROM [dbo].[Solution] WHERE [Id] =  @ParentSolutionId)
+		THROW 51000, 'Parent Solution record does not exist.', 1;  
+
+	SELECT @PublishedStatusId=PublishedStatusId, @AuthorityStatusId = AuthorityStatusId, @SupplierStatusId = SupplierStatusId
+		FROM [dbo].[Solution] WHERE [Id] =  @ParentSolutionId;
+
+	/***
+	Add the Addition Service if not already present
+	***/
+	IF NOT EXISTS (SELECT 1 FROM [dbo].[Solution] WHERE [Id] =  @AdditionServiceId)	
+	INSERT INTO [dbo].[Solution]
```

Comment:
Is a Version Required?

Reply:
Version not required. The version is information field defined by the supplier for the parent solution. Two versions of a solution will have different solution ids.